src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -0,0 +1,184 @@
+﻿using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Parse a project.json, and add additional dependencies, then write a out new project.json.
+    /// </summary>
+    public class AddDependenciesToProjectJson : Task
+    {
+        private Regex VersionStructureRegex = new Regex(@"(\d+\.\d+\.\d+)-((beta|rc2|rc3)?-?(\d+))");
```

Comment:
I think it will work for our current packages but might not moving forward and we should make ti configurable. It does look like you are missing a "?" at the end because the BuildNumberMinor is still optional and doesn't exist yet.